src/lib/net/address.c

```diff
@@ -773,8 +773,12 @@ tor_addr_is_null(const tor_addr_t *addr)
 
   switch (tor_addr_family(addr)) {
     case AF_INET6: {
-      uint32_t *a32 = tor_addr_to_in6_addr32(addr);
-      return (a32[0] == 0) && (a32[1] == 0) && (a32[2] == 0) && (a32[3] == 0);
+      const char *ipv6 = (const char *)tor_addr_to_in6_addr8(addr);
+      int result = 1, i;
+      for (i = 0; i < 16; ++i) {
+        result &= ipv6[i] == 0;
+      }
+      return result;
```

Comment:
I think this code does exactly the same thing as the previous code.

If you think it does something different, please provide a test case that produces a different result on the old and new code.

If you think it is the same, please don't modify this code.

In general, we try to change one function per commit or pull request. That way, we know why the tests are failing.

Reply:
We also try to minimise the changes, so they are easy to review,